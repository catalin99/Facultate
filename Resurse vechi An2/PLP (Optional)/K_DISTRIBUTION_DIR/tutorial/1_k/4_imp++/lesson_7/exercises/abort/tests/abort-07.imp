// This program prints 3 messages in a random order. There are 3! = 6 possible
// orders for the messages. If we replace the first halt with an abort, then
// only the first message is guaranteed to printed, and the number of possible
// messsage orders becomes 11.

spawn { print("Hi from thread 1!\n"); halt; };
spawn { print("Hi from thread 2!\n"); halt; };
spawn { print("Hi from thread 3!\n"); halt; };
