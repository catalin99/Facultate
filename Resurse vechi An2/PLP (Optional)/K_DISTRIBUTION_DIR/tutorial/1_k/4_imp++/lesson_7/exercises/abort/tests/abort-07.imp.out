Search results:

Solution 1:
S:String -->
"Hi from thread 1!\nHi from thread 2!\nHi from thread 3!\n"

Solution 2:
S:String -->
"Hi from thread 1!\nHi from thread 3!\nHi from thread 2!\n"

Solution 3:
S:String -->
"Hi from thread 2!\nHi from thread 1!\nHi from thread 3!\n"

Solution 4:
S:String -->
"Hi from thread 2!\nHi from thread 3!\nHi from thread 1!\n"

Solution 5:
S:String -->
"Hi from thread 3!\nHi from thread 1!\nHi from thread 2!\n"

Solution 6:
S:String -->
"Hi from thread 3!\nHi from thread 2!\nHi from thread 1!\n"
